### DISCUSSION TITLE
Terminal window(s) as tabs inside editor

### DISCUSSION THREAD

**Discussion by juliuskoskela** [lapce/lapce#1176] [Ideas]
Most of the existing editors (Lapce included) spawn the terminal as a separate panel taking space either at the bottom or either sides of the editor* window (last I tried I couldn't find an option to move the terminal to the right from Lapce, maybe this has not been implemented yet?). I have always found this cumbersome. At the bottom the terminal window is too wide, but not tall enough and taking space from the editor. I prefer it to the right of the editor to have more vertical space, but I've always thought it would be much more convenient if the terminal would actually be one of the tabs in the editor itself. 

Benefits include:

- Being able to move and resize the terminal as part of the editor layout.
- Being able to hide the terminal as one of the tabs in the editor.
- When using the terminal have a bigger vertical view by default.
- Less splitting of the overall panel layout, always have maximum width for open files/terminals.
- A test could spawn a dedicated terminal instance and if run again spawn a new one, making inspecting differences in output more convenient.
- Relevant terminal windows could be grouped together contextually with files they relate to using windowing in the editor.

Just a rough idea that's been floating in my mind when using Vscode. Interested if anyone else would see benefit in it. Lapce looks absolutely amazing and I can feel the difference in performance very clearly compared to Vscode. Everything feels almost instantaneous. Great job!

*I might be using the wrong nomenclature. By editor I mean the area in which the file tabs are opened and edited in and which allows for splitting of the view, excluding the terminal panel or any side panel(s).

**Comment by ghishadow** [comment]
@juliuskoskela You can already drag terminal to the right, but making it like tab is good idea
![image](https://user-images.githubusercontent.com/9583775/189796726-dee83623-32df-4131-87eb-3a345c29f92e.png)
